index.d.ts

```diff
@@ -0,0 +1,183 @@
+type OnCanvasDrawFunction = (instance: AudioMotionAnalyzer) => unknown;
+type OnCanvasResizeFunction = (
+  reason: CanvasResizeReason,
+  instance: AudioMotionAnalyzer
+) => unknown;
+
+type CanvasResizeReason = "create" | "fschange" | "lores" | "resize" | "user";
+
+export interface Options {
+  audioCtx?: AudioContext;
```

Comment:
`audioCtx` and `source` are only valid in the constructor call, but this interface will be used by the `setOptions()` method as well.
Can we remove them from this interface and create another one for the constructor definition, say `ConstructorOptions`, which extends this and adds those two properties?